src/main/java/org/ambraproject/rhino/service/impl/ArticleCrudServiceImpl.java

```diff
@@ -526,40 +526,93 @@ public Article readArticle(ArticleIdentifier articleIdentifier) {
+          "FROM ArticleRevision ar, ArticleFile file " +
+          "INNER JOIN ar.ingestion ai " +
+          "INNER JOIN ar.ingestion.article at " +
+          "WHERE ai.publicationDate >= :fromDate AND ai.publicationDate <= :toDate " +
+          "AND ar.revisionId IS NOT NULL " +
+          "AND file.ingestion = ai " +
+          "AND file.bucketName = :bucketName";
+      Query query = session.createQuery(queryString);
       query.setParameter("fromDate", java.sql.Date.valueOf(fromDate));
       query.setParameter("toDate", java.sql.Date.valueOf(toDate));
+      if (bucketName != null) {
+        query.setParameter("bucketName", bucketName);
+      }
       return (Collection<ArticleRevision>) query.list();
     });
   }
 
   @Override
   public Collection<ArticleRevision> getArticlesRevisedOn(LocalDate fromDate, LocalDate toDate) {
+    return getArticlesRevisedOn(fromDate, toDate, null);
+  }
+
+  @Override
+  public Collection<ArticleRevision> getArticlesRevisedOn(LocalDate fromDate, LocalDate toDate, String bucketName) {
     return hibernateTemplate.execute(session -> {
-      Query query = session.createQuery("" +
-          "SELECT ar " +
+      String queryString = bucketName == null
+          ? "SELECT ar " +
           "FROM ArticleRevision ar " +
           "INNER JOIN ar.ingestion ai " +
           "INNER JOIN  ar.ingestion.article at " +
           "WHERE ai.revisionDate >= :fromDate AND ai.revisionDate <= :toDate " +
-          "AND ar.revisionId IS NOT NULL");
+          "AND ar.revisionId IS NOT NULL"
+          : "SELECT ar " +
+          "FROM ArticleRevision ar, ArticleFile file " +
+          "INNER JOIN ar.ingestion ai " +
+          "INNER JOIN ar.ingestion.article at " +
+          "WHERE ai.revisionDate >= :fromDate AND ai.revisionDate <= :toDate " +
+          "AND ar.revisionId IS NOT NULL " +
+          "AND file.ingestion = ai " +
+          "AND file.bucketName = :bucketName";
+      Query query = session.createQuery(queryString);
       query.setParameter("fromDate", java.sql.Date.valueOf(fromDate));
       query.setParameter("toDate", java.sql.Date.valueOf(toDate));
+      if (bucketName != null) {
+        query.setParameter("bucketName", bucketName);
+      }
       return (Collection<ArticleRevision>) query.list();
     });
   }
 
   @Override
-  public void updateIsPreprintOfDoi(ArticleIngestionIdentifier articleId, String preprintOfUrl) throws IOException {
+  public Collection<ArticleRevision> getPreprintArticlesWithoutVor() {
+    return hibernateTemplate.execute(session -> {
+      String subQuery = "" +
+          "SELECT DISTINCT fi.ingestion " +
+          "FROM ArticleFile fi " +
+          "INNER JOIN fi.ingestion ai " +
+          "WHERE fi.bucketName = 'preprints' " +
+          "AND ai.isPreprintOfDoi IS NULL";
+      String queryString = "" +
+          "SELECT DISTINCT ar " +
+          "FROM ArticleRevision ar " +
+          "WHERE ar.ingestion IN (" + subQuery + ")";
```

Comment:
I usually prefer using `EXISTS` rather than `IN`, as EXISTS will perform better.  Though, I don't know how to translate EXISTS to an HQL equivalent.

Reply:
This is also from master and don't know how to test this and use of exists in hql. will prefer leaving it like that